peekaboo/ruleset/rules.py

```diff
@@ -224,22 +225,42 @@ def evaluate(self, sample):
                            True)
 
 
-class OfficeMacroWithAutoActionRule(Rule):
+class OfficeMacroWithSuspiciousKeyword(Rule):
     """ A rule checking the sample for Office macros. """
-    rule_name = 'office_macro_with_auto_action'
+    rule_name = 'office_macro_with_suspicious_keyword'
+
+    def get_config(self):
+        # get list of keywords from config file
+        suspicious_keyword_list = self.get_config_value(
+            'keyword', [])
+        # convert into list of regexes
+        self.suspicious_keywords = []
+        # config.get_by_type does not support flags
+        # so we compile ourself
+        try:
+            for r in suspicious_keyword_list:
+                self.suspicious_keywords.append(re.compile('.*'+r+'.*', re.IGNORECASE))
+        except (ValueError, TypeError) as error:
+            raise Exception(
+                'Failed to compile regular expression "%s" (section %s, '
+                'option %s): %s' % (r, self.rule_name, 'keyword', error))
+        if not self.suspicious_keywords:
+            raise PeekabooRulesetConfigError(
+                "Empty suspicious keyword list, check %s rule config." %
+                self.rule_name)
 
     def evaluate(self, sample):
         """ Report the sample as bad if it contains a macro. """
-        if sample.office_macros_with_auto_action:
+        if has_office_macros_with_suspicious_keyword(sample.file_path, sample.file_extension, self.suspicious_keywords):
```

Comment:
This eliminates the redirect (and caching) via Sample but still leaved the decision making up to `has_office_macros_with_suspicious_keyword()`. Why not have a `Sample.office_macro_names()` that extracts the macro names (and perhaps other stuff, aka `Sample.active_office_content()`) and walk through that "report" here in the rule?

Reply:
90% there with latest changes. Let's merge it like that. If we ever want to really polish it, we can consider moving the evaluation logic from `report.has_office_macros_with_suspicious_keyword()` and `report.has_office_macros()` into the respective rules to make the analyzer module even more generic.